src/Dependencies/Collections/Extensions/IEnumerableExtensions.cs

```diff
@@ -349,25 +377,34 @@ public static T[] AsArray<T>(this IEnumerable<T> source)
 
         public static ImmutableArray<TResult> SelectAsArray<TSource, TResult>(this IEnumerable<TSource>? source, Func<TSource, TResult> selector)
```

Comment:
Extensions that work on null receiver are odd.

Reply:
don't disagree.  however, a *lot* of collect/ienumrable methods do this.  so this is keeping things mostly consistent.  would prefer to not rock the boat there for now.